File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */

 function toArray(list){
    let arr = [];
    while(list){
        arr.push(list.val);
        list = list.next
    }
    return arr;
 }

 function toList(arr){
    let curNode = new ListNode(0);
    let dummy = curNode;
    for(let i = 0; i < arr.length; i++){
        curNode.next = new ListNode(arr[i]);
        curNode = curNode.next;
    }
    return dummy.next
 }

function addTwoNumbers(l1, l2) {
    const arr1 = toArray(l1); 
    const arr2 = toArray(l2);
    const maxLength = Math.max(arr1.length,arr2.length)
    let sum = 0;
    let carry = 0;
    let sumArr = []; 

for(let i = 0; i < maxLength; i++){
        sum = (arr1[i] || 0) + (arr2[i] || 0) + carry;
        carry = sum >= 10 ? 1 : 0;
        sumArr.push(sum % 10);
    }
    if(carry){
        sumArr.push(carry)
    }
  return toList(sumArr)
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        dummyNode = ListNode()
        cur = dummyNode

        while l1 or l2 or carry:
            sumVal = carry

            if l1:
                sumVal += l1.val
                l1 = l1.next
            if l2:
                sumVal += l2.val
                l2 = l2.next
            carry = sumVal // 10
            cur.next = ListNode(sumVal % 10)
            cur = cur.next
        return dummyNode.next
             